spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,252 @@
+      expect {
+        reservation_manager.create_block(room_ids: [3],
+                                         check_in_date: "2019-03-10",
+                                         check_out_date: "2019-03-15",
+                                         discount_rate: 0.10)
+      }.must_raise ArgumentError
+    end
+  end
+
+  describe "check_available_rooms_in_blocks" do
+    it "can get available rooms in a given block" do
+      reservation_manager.create_block(room_ids: [1, 2, 3],
+                                       check_in_date: "2019-03-10",
+                                       check_out_date: "2019-03-15",
+                                       discount_rate: 0.10)
+      expect(reservation_manager.check_available_rooms_in_blocks(block_id: 1).length).must_equal 3
+    end
+
+    it "returns an empty array if no room is available in a block" do
+      reservation_manager.create_block(room_ids: [2, 3],
+                                       check_in_date: "2019-03-10",
+                                       check_out_date: "2019-03-15",
+                                       discount_rate: 0.10)
+
+      reservation_manager.reserve_from_block(room_id: 2, block_id: 1)
+      reservation_manager.reserve_from_block(room_id: 3, block_id: 1)
+      expect(reservation_manager.check_available_rooms_in_blocks(block_id: 1).length).must_equal 0
+    end
+
+    it "raises ArgumentError if no block is found" do
+      expect { reservation_manager.check_available_rooms_in_blocks(1) }.must_raise ArgumentError
+    end
+  end
+
+  describe "reserve_from_block" do
+    it "adds to the reservation list after reserving from the block" do
+      reservation_manager.create_block(room_ids: [1, 2, 3],
+                                       check_in_date: "2019-03-10",
+                                       check_out_date: "2019-03-15",
+                                       discount_rate: 0.10)
+      before_reserve = reservation_manager.reservations.length
+      reservation_manager.reserve_from_block(room_id: 1, block_id: 1)
+      after_reserve = reservation_manager.reservations.length
+      expect(after_reserve - before_reserve).must_equal 1
+    end
+
+    it "changes the room's status in the block to unavailable after the room has been reserved from the block" do
+      reservation_manager.create_block(room_ids: [1, 2, 3],
+                                       check_in_date: "2019-03-10",
+                                       check_out_date: "2019-03-15",
+                                       discount_rate: 0.10)
+      reservation_manager.reserve_from_block(room_id: 1, block_id: 1)
+      block = reservation_manager.blocks.find { |block| block.block_id == 1 }
+      room_status = block.rooms_info.each do |current_room_id, status|
+        return status if current_room_id == 1
+      end
+      expect(room_status).must_equal :UNAVAILABLE
+    end
+  end
+
+  describe "validate_id method" do
```

Comment:
`validate_id` seems like it should be a private helper method and thus not tested directly.